repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -423,8 +423,8 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         } catch (URISyntaxException use) {
             LOG.debug("{}:{} -- Invalid URI requested: {}", clusterId, nodeId, wrappedRequest.getRequestURI(), use);
             wrappedResponse.sendError(HttpServletResponse.SC_BAD_REQUEST, "Error processing request");
-        } catch (Exception ex) {
-            LOG.error("{}:{} -- Exception encountered while processing filter chain.", clusterId, nodeId, ex);
+        } catch (Throwable t) {
```

Comment:
I'm wondering if we should handle `Exception`s and `Error`s separately, and rethrow `Error`s. That way we can still log the issue and set the status code, but we aren't claiming to have handled the `Error` in question by swallowing it. It feels a little sketchy to just swap `Exception` for `Throwable` in cases where the JVM runs out of memory or something even more strange.

Sonar may have a point on this one. While it makes sense for us to catch a `NoClassDefFoundError` (or maybe, more generally, a `LinkageError`) in certain cases, does it really make sense to catch `Throwable` here? What if the JVM hits  

https://sbforge.org/sonar/rules/show/squid:S1181?layout=false

Reply:
so rethrowing the error won't solve anything, because the finally block will still trigger

Reply:
or are you suggesting we catch it, log and set the response then rethrow it?

Reply:
Right, the ladder. Catch it, log it, set it, throw it. Bop it.

Reply:
I concur with the latter as well. Re-Throwing definitely makes sure the JVM is able to clean up after itself if it so chooses.